kie-wb-tests/kie-wb-tests-rest/src/main/java/org/kie/wb/test/rest/client/RestWorkbenchClient.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.wb.test.rest.client;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+import org.guvnor.rest.client.AddRepositoryToOrganizationalUnitRequest;
+import org.guvnor.rest.client.CompileProjectRequest;
+import org.guvnor.rest.client.CreateOrCloneRepositoryRequest;
+import org.guvnor.rest.client.CreateOrganizationalUnitRequest;
+import org.guvnor.rest.client.CreateProjectRequest;
+import org.guvnor.rest.client.DeleteProjectRequest;
+import org.guvnor.rest.client.DeployProjectRequest;
+import org.guvnor.rest.client.InstallProjectRequest;
+import org.guvnor.rest.client.JobRequest;
+import org.guvnor.rest.client.JobResult;
+import org.guvnor.rest.client.JobStatus;
+import org.guvnor.rest.client.OrganizationalUnit;
+import org.guvnor.rest.client.ProjectRequest;
+import org.guvnor.rest.client.ProjectResponse;
+import org.guvnor.rest.client.RemoveOrganizationalUnitRequest;
+import org.guvnor.rest.client.RemoveRepositoryFromOrganizationalUnitRequest;
+import org.guvnor.rest.client.RemoveRepositoryRequest;
+import org.guvnor.rest.client.RepositoryRequest;
+import org.guvnor.rest.client.RepositoryResponse;
+import org.guvnor.rest.client.TestProjectRequest;
+import org.guvnor.rest.client.UpdateOrganizationalUnit;
+import org.guvnor.rest.client.UpdateOrganizationalUnitRequest;
+import org.kie.wb.test.rest.util.HttpRequestFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestWorkbenchClient implements WorkbenchClient {
```

Comment:
I was looking at [Resteasy Client API](http://docs.jboss.org/resteasy/docs/3.0.17.Final/userguide/html_single/index.html#RESTEasy_Client_Framework) when I was rewriting our framework. Unfortunately, we depend on RESTEasy 2.x which does not contain this feature. So unless we update to RESTEasy 3.x, it will not be possible to make use of it. But even if we switch to it, we will still need some kind of wrapper around it like we have right now, because otherwise there will be a lot of duplicated code in our tests.